### PR TITLE
Changes

### PR DESCRIPTION
**Pull request by tokevinm**
Summary of most recent changes:
- Changed CoinGecko API endpoint to allow simplified additions to user options
- Formatted received data into a nested dictionary to make more easily accessible
- Reorganized few sections of code to get rid of redundance
- Restructured sections of code to be more readable 
- Assets with a price under $1 are now formatted to 4 decimal places
- Minor email formatting for the case of no user choices

### CODE REVIEW

**Comment by tokevinm** [comment]
Not sure why the "Files changed" shows everything changed

File: main.py [hunk 1]
```diff
@@ -1,61 +1,53 @@
...
+else:
+    close_significance = "Daily"
+
+for user in users_data:
+    user_email = user["pleaseEnterYourEmailAddress:"]
+    user_options = user.get("anyExtraDataYou'dLikeInYourReport?", None)
+    btc_up_down = daily_up_down_icon(asset_data["bitcoin"]["24h_change_percent"])
+    message_body = (f"Subject:BTC {close_significance} Close: "
+                    f"{asset_data["bitcoin"]["price"]} {btc_up_down}{asset_data["bitcoin"]["24h_change_percent"]}%\n\n")
+    if user_options is not None:
```
Comment (michaelshum321) [inline main.py:42]: I usually just do `if user_options` and it's the same

File: main.py [hunk 2]
```diff
@@ -1,61 +1,53 @@
...
+        user_choices = user_options.split()[::2]
+        for choice in user_choices:
+            asset_lower = choice.lower()
+            asset_up_down = daily_up_down_icon(asset_data[asset_lower]["24h_change_percent"])
+            message_add = (f"{asset_data[asset_lower]["ticker_upper"]}: {asset_data[asset_lower]["price"]} "
+                           f"{asset_up_down}{asset_data[asset_lower]["24h_change_percent"]}%\n")
+            message_body += message_add
```
Comment (michaelshum321) [inline main.py:49]: `message_add` is only used here right? might as well just do `messagebody += (f""...)

File: asset_data.py [hunk 3]
```diff
@@ -1,67 +1,89 @@
...
+    """Requests, manages, and formats data received from the CoinGecko API"""
+    def __init__(self):
+
+        self.cg_endpoint = "https://api.coingecko.com/api/v3"
+        self.cg_assets_list = [
+            "bitcoin",
+            "ethereum",
+            "solana",
+            "dogecoin"
+        ]
```
Comment (michaelshum321) [inline asset_data.py:17]: nice update over the last

**Review by michaelshum321 — APPROVED** [review]
yea everything here looks pretty reasonable

File: main.py (hunk 2 above)
Comment (tokevinm) [reply main.py:49]: Ooooo yea before this update I had a different message_add for each diff crypto but didn't catch that after consolidating 😗 tyty

File: asset_data.py [hunk 4]
```diff
@@ -1,67 +1,89 @@
...
+        self.cg_assets_data = {}
+        self.cg_global_data = {}
+        self.crypto_mcap_perc_total = None
+        self.crypto_total_mcap = None
+        # crypto_mcap_perc is a dictionary of top10 assets and percentages of total market cap
```
Comment (tokevinm) [inline asset_data.py:22]: Change comment with variable or just remove completely? Saved data not in use yet but still considering it

File: main.py [hunk 5]
```diff
@@ -1,61 +1,53 @@
...
+    else:
+        icon = ""
+    return icon
+
+
+day_of_month = datetime.now().strftime("%d")
+day_of_week = datetime.now().strftime("%w")
+if day_of_month == "01":
+    close_significance = "MONTHLY"
+elif day_of_week == "0":
```
Comment (tokevinm) [inline main.py:31]: Mistake: Weekly closes are Sunday night/Monday 00:00 UTC (Should be "day_of_week == 1") if 

"%w is Weekday as a number 0-6, 0 is Sunday" 

https://www.w3schools.com/python/python_datetime.asp

File: asset_data.py (hunk 4 above)
Comment (michaelshum321) [reply asset_data.py:22]: you can leave it as a TODO cmt, aka `# TODO: consider adding cryptp_mcap_perc - a dict of top 10 assets...`

**Pull request merged**